sycl/source/detail/scheduler/commands.cpp

```diff
@@ -2503,7 +2502,7 @@ pi_int32 enqueueImpKernel(
       sycl::detail::pi::PiKernel FoundKernel = nullptr;
       std::tie(FoundKernel, KernelMutex, EliminatedArgMask, std::ignore) =
           detail::ProgramManager::getInstance().getOrCreateKernel(
-              ContextImpl, DeviceImpl, KernelName, SyclProg.get());
```

Comment:
This should be unreachable, `program_impl::is_cacheable()` is true iff it was built with `build_with_kernel_name`, which is being removed. I think we should get rid of `is_cacheable` and all branches that check it to be true as well.

Reply:
removed the unreachable branch.